### PR TITLE
make the template processing more compatible with the dynamic client

### PR DESCRIPTION
**Pull request by deads2k**
Simplify processing a template via the API and getting a result you can use with dynamic client.

post-rebase request from @mfojtik 

/assign @mfojtik

### CODE REVIEW

File: pkg/client/templateprocessing/dynamic_process.go [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+	if err != nil {
+		return nil, err
+	}
+
+	return c.ProcessToListFromUnstructured(&unstructured.Unstructured{Object: unstructuredTemplate})
+}
+
+func (c *dynamicTemplateProcessor) ProcessToListFromUnstructured(unstructuredTemplate *unstructured.Unstructured) (*unstructured.UnstructuredList, error) {
+	processedTemplate, err := c.client.Resource(templatev1.GroupVersion.WithResource("processedtemplates")).
+		Namespace("default").Create(unstructuredTemplate)
```
Comment (mfojtik) [inline pkg/client/templateprocessing/dynamic_process.go]: can you check the error here?

File: pkg/client/templateprocessing/dynamic_process.go [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+	// convert the template into something we iterate over as a list
+	if err := unstructured.SetNestedField(processedTemplate.Object, processedTemplate.Object["objects"], "items"); err != nil {
+		return nil, err
+	}
+	processedList, err := processedTemplate.ToList()
+	if err != nil {
+		return nil, err
+	}
+	return processedList, nil
```
Comment (mfojtik) [inline pkg/client/templateprocessing/dynamic_process.go]: i would just return `processedTemplate.ToList()`

File: pkg/project/registry/projectrequest/delegated/delegated.go [hunk 3]
```diff
@@ -155,21 +156,8 @@ func (r *REST) Create(ctx context.Context, obj runtime.Object, createValidation
...
-	}
-	processedTemplate, err := r.client.Resource(schema.GroupVersionResource{Group: "template.openshift.io", Version: "v1", Resource: "processedtemplates"}).
-		Namespace("default").Create(&unstructured.Unstructured{Object: unstructuredTemplate})
-	// convert the template into something we iterate over as a list
-	if err := unstructured.SetNestedField(processedTemplate.Object, processedTemplate.Object["objects"], "items"); err != nil {
-		return nil, err
-	}
-	processedList, err := processedTemplate.ToList()
+	templateProcessor := templateprocessing.NewDynamicTemplateProcessor(r.client)
+	processedList, err := templateProcessor.ProcessToList(template)
```
Comment (mfojtik) [inline pkg/project/registry/projectrequest/delegated/delegated.go]: nit: `templateprocessing.NewDynamicTemplateProcessor(r.client).ProcessToList(template)` but I don't care much

File: pkg/template/templateprocessing/template_test.go [hunk 4]
```diff
@@ -453,3 +434,13 @@ func TestProcessTemplateParameters(t *testing.T) {
 		t.Errorf("unexpected output: %s", diff.StringDiff(string(exp), string(result)))
 	}
 }
+
+// AddParameter adds new custom parameter to the Template. It overrides
```
Comment (mfojtik) [inline pkg/template/templateprocessing/template_test.go]: nit: lower case

**Comment by mfojtik** [comment]
@deads2k small nits, LGTM otherwise

**Comment by deads2k** [comment]
comments addressed.

**Pull request merged**